agent/config/config.go

```diff
@@ -86,15 +86,28 @@ func Parse(data string, format string) (c Config, err error) {
 		"watches",
 	})
 
-	// 	toJSON := func(v interface{}) string {
-	// 		b, err := json.MarshalIndent(v, "", "    ")
-	// 		if err != nil {
-	// 			panic(err)
-	// 		}
-	// 		return string(b)
-	// 	}
-	// 	fmt.Println("raw:", toJSON(raw))
-	// 	fmt.Println("patched:", toJSON(m))
+	// There is a difference of representation of some fields depending on
+	// where they are used. The HTTP API uses CamelCase whereas the config
+	// files use snake_case and between the two there is no automatic mapping.
+	// While the JSON and HCL parsers match keys without case (both `id` and
+	// `ID` are mapped to an ID field) the same thing does not happen between
+	// CamelCase and snake_case. Since changing either format would break
+	// existing setups we have to support both and slowly transition to one of
+	// the formats. Also, there is at least one case where we use the "wrong"
+	// key and want to map that to the new key to support deprecation
+	// (`check.id` vs `service.check.CheckID`) See [GH-3179]. TranslateKeys
+	// maps potentially CamelCased values to the snake_case that is used in the
+	// config file parser. If both the CamelCase and snake_case values are set,
+	// the snake_case value is used and the other value is discarded.
+	TranslateKeys(m, map[string]string{
```

Comment:
Do we need to account for the `ScriptArgs` member in `CheckType` as well? In the API structure that got mapped to `args` but is probably broken for config files.

Reply:
`ScriptArgs` isn't even documented on the API: https://www.consul.io/api/agent/check.html

I'll add a map for `scriptargs` and `script_args`.

Reply:
Done

Reply:
Added some documentation in d3cebace7c2f4fc61beb207a037cbb44daefd9a8.